src/tests/unit_tests/test_vectorstores.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest.mock import Mock
+
+import pytest
+from google.cloud import firestore  # type: ignore
+from google.cloud.firestore import CollectionReference  # type: ignore
+from langchain_community.embeddings import FakeEmbeddings
+
+from langchain_google_firestore.vectorstores import FirestoreVectorStore
+
+TEST_COLLECTION = "test_collection"
+
+
+def get_embeddings():
+    return FakeEmbeddings(size=100)
+
+
+def get_client():
+    return firestore.Client()
+
+
+@pytest.fixture(scope="session", autouse=True)
+def cleanup_firestore():
+    client = get_client()
+    for doc in client.collection(TEST_COLLECTION).stream():
+        doc.reference.delete()
+
+
+def test_firestore_vectorstore_initialization():
+    """
+    Tests FirestoreVectorStore initialization with mocked embeddings,
+    focusing on correct attribute setting and potential errors.
+
+    This test uses `unittest.mock` and manual patching.
+    """
+
+    # Mock Embeddings class and its attributes
+    mocked_embeddings = Mock()
+
+    # Create FirestoreVectorStore instance
+    firestore_store = FirestoreVectorStore("my_collection", mocked_embeddings)
+
+    # Assertions to verify attribute values and error handling
+    assert isinstance(firestore_store.collection, CollectionReference)
+    assert firestore_store.embeddings == mocked_embeddings
+
+
+def test_firestore_add_vectors():
+    """
+    An end-to-end test for adding vectors to FirestoreVectorStore.
+    """
+    # Create FirestoreVectorStore instance
+    firestore_store = FirestoreVectorStore(
+        TEST_COLLECTION, get_embeddings(), client=get_client()
+    )
+
+    # Add vectors to Firestore
+    firestore_store.add_texts(
+        ["test_doc1", "test_doc2"],
+        metadatas=[{"foo": "bar"}, {"baz": "qux"}],
+        ids=["1", "2"],
+    )
+
+    # Verify that the vectors were added to Firestore
+    docs = firestore_store.collection.stream()
+    for doc in docs:
```

Comment:
Can we instead create the list of expected LangChain Documents and compare them? -- if it's only 2 docs. Then use get

To compare lists of objects you could use
```
test_case = TestCase()
test_case.assertCountEqual(expected_docs, docs_after_write)
```

See test_document_loader.py for reference of TestCase


Same for the rest of tests

Reply:
Done